### PR TITLE
global: user hub UI

### PR DESCRIPTION
**Pull request by mvesper**
* Adds the first implementation of the 'user hub' user interface.

### CODE REVIEW

**Comment by jirikuncar** [comment]
@mvesper can you mark strings for translations in Angular templates with `translate` attribute? (see  e.g. https://github.com/inveniosoftware/invenio-search-ui/pull/40/files#diff-63ef883cfeb6fb0541e35aa985bb9b6dR3)

File: invenio_circulation/static/js/circulation/user_hub_app.js [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+ * 59 Temple Place, Suite 330, Boston, MA 02111-1307, USA.
+ *
+ * In applying this license, CERN does not
+ * waive the privileges and immunities granted to it by virtue of its status
+ * as an Intergovernmental Organization or submit itself to any jurisdiction.
+ */
+
+
+require([
+    'node_modules/angular/angular.js',
```
Comment (drjova) [inline invenio_circulation/static/js/circulation/user_hub_app.js]: You can remove the `.js` :)

File: invenio_circulation/bundles.py [hunk 2]
```diff
@@ -58,3 +58,18 @@
         'invenio-search-js': '~0.2.0',
     }
 )
+
+user_hub_js = NpmBundle(
+    'node_modules/requirejs/require.js',
+    'node_modules/angular/angular.js',
```
Comment (drjova) [inline invenio_circulation/bundles.py]: I see that you use `requirejs` to include angular you can remove it from here :)

File: invenio_circulation/bundles.py [hunk 3]
```diff
@@ -58,3 +58,18 @@
         'invenio-search-js': '~0.2.0',
     }
 )
+
+user_hub_js = NpmBundle(
+    'node_modules/requirejs/require.js',
```
Comment (drjova) [inline invenio_circulation/bundles.py]: IMHO we don't need it here

**Review by drjova — CHANGES_REQUESTED** [review]

**Review by drjova — APPROVED** [review]

File: invenio_circulation/receivers.py [hunk 4]
```diff
@@ -66,6 +67,7 @@ def run(self, event):
             data, _ = self.circulation_event_schema.dump(data)
             self._run(item, data)
             item.commit()
+            RecordIndexer().index(item)
```
Comment (egabancho) [inline invenio_circulation/receivers.py:70]: I would move this changes to a different commit, it might help in the future

Comment (mvesper) [reply invenio_circulation/receivers.py:70]: done :)
https://github.com/inveniosoftware/invenio-circulation/pull/60

**Comment by egabancho** [comment]
Once you rebase on top of https://github.com/inveniosoftware/invenio-circulation/pull/60 ... LGTM

**Comment by david-caro** [comment]
Looks good to me too

File: examples/app.py [hunk 5]
```diff
@@ -79,6 +79,12 @@ def init_app(app):
             '/hooks/receivers/circulation_request/events/'),
         CIRCULATION_ACTION_RETURN_URL=(
             '/hooks/receivers/circulation_return/events/'),
+        CIRCULATION_ACTION_EXTEND_URL=(
+            '/hooks/receivers/circulation_extend/events/'),
+        CIRCULATION_ACTION_LOSE_URL=(
+            '/hooks/receivers/circulation_lose/events/'),
+        CIRCULATION_ACTION_CANCEL_URL=(
+            '/hooks/receivers/circulation_cancel/events/')
```
Comment (jirikuncar) [inline examples/app.py]: can you add a comma at the end of line so we have a clean diff next time? (see line 81 in current diff 😉 )

File: invenio_circulation/static/js/circulation/directives/circulationUserHoldings.js [hunk 6]
```diff
@@ -0,0 +1,114 @@
...
+
+        $http({
+          method: 'POST',
+          url: attributes.extendEndpoint,
+          headers: {
+            'Content-Type': 'application/json'
+          },
+          data: data,
+        }).then(function(response) {
+        });
```
Comment (jirikuncar) [inline invenio_circulation/static/js/circulation/directives/circulationUserHoldings.js:75]: What about the empty response handler?

**Pull request merged**